ext/drreg/drreg.c

```diff
@@ -2069,7 +2069,12 @@ drreg_event_restore_state_without_ilist(void *drcontext, bool restore_memory,
                         __FUNCTION__, pc);
                 }
             }
-        } else if (instr_get_opcode(&inst) == IF_X86_ELSE(OP_lahf, OP_mrs)) {
+        } else if (instr_get_opcode(&inst) ==
+                   IF_X86_ELSE(OP_lahf, IF_RISCV64_ELSE(OP_hint, OP_mrs))) {
```

Comment:
I don't understand why this would look for the HINT opcode and assume it meant a condition code load?  HINT looks like a non-arch-state-affecting opcode which a condition code load would not be.
Plus, RISC-V has no condition codes?  So drreg would never spill them.  So RISC-V should never go into this code block?  Add a local helper function `instr_is_aflags_spill()` which always returns false for RISC-V?

Ditto below.

Reply:
This was because I made a fast compilation hack. I'll add aflags spill/restore functions as you suggested.

Reply:
Done.